rules/aws_iam_role_deprecated_policy_attributes.go

```diff
@@ -0,0 +1,80 @@
+package rules
+
+import (
+	"github.com/terraform-linters/tflint-plugin-sdk/hclext"
+	"github.com/terraform-linters/tflint-plugin-sdk/tflint"
+	"github.com/terraform-linters/tflint-ruleset-aws/project"
+)
+
+// AwsIAMRoleDeprecatedPolicyAttributesRule checks that neither `inline_policy` or `managed_policy_arns` are used in an `aws_iam_role`
+type AwsIAMRoleDeprecatedPolicyAttributesRule struct {
+	tflint.DefaultRule
+
+	resourceType string
+}
+
+// NewAwsIAMRoleDeprecatedPolicyAttributesRule returns new rule with default attributes
+func NewAwsIAMRoleDeprecatedPolicyAttributesRule() *AwsIAMRoleDeprecatedPolicyAttributesRule {
+	return &AwsIAMRoleDeprecatedPolicyAttributesRule{
+		resourceType: "aws_iam_role",
+	}
+}
+
+// Name returns the rule name
+func (r *AwsIAMRoleDeprecatedPolicyAttributesRule) Name() string {
+	return "aws_iam_role_deprecated_policy_attributes"
+}
+
+// Enabled returns whether the rule is enabled by default
+func (r *AwsIAMRoleDeprecatedPolicyAttributesRule) Enabled() bool {
+	// `inline_policy` and `managed_policy_arns` were deprecated in 5.68.0 and 5.72.0 respectively
+	// At time of writing, this is quite recent, so don't automatically enable the rule.
+	return false
+}
+
+// Severity returns the rule severity
+func (r *AwsIAMRoleDeprecatedPolicyAttributesRule) Severity() tflint.Severity {
+	return tflint.WARNING
+}
+
+// Link returns the rule reference link
+func (r *AwsIAMRoleDeprecatedPolicyAttributesRule) Link() string {
+	return project.ReferenceLink(r.Name())
+}
+
+// Check checks that aws_iam_role resources don't have any `inline_policy` blocks or the `managed_policy_arns` attribute
+func (r *AwsIAMRoleDeprecatedPolicyAttributesRule) Check(runner tflint.Runner) error {
+	resources, err := runner.GetResourceContent(r.resourceType, &hclext.BodySchema{
+		Blocks: []hclext.BlockSchema{
+			{
+				Type: "inline_policy",
+			},
+		},
+		Attributes: []hclext.AttributeSchema{
+			{
+				Name: "managed_policy_arns",
+			},
+		},
+	}, nil)
+	if err != nil {
+		return err
+	}
+
+	for _, resource := range resources.Blocks {
+		attribute, exists := resource.Body.Attributes["managed_policy_arns"]
+
+		if exists {
+			if err := runner.EmitIssue(r, "The managed_policy_arns argument is deprecated. Use the aws_iam_role_policy_attachment resource instead. If Terraform should exclusively manage all managed policy attachments (the current behavior of this argument), use the aws_iam_role_policy_attachments_exclusive resource as well.", attribute.Range); err != nil {
```

Comment:
I would flip these. Users should prefer the `exclusive` resources by default and fall back on the non-authoritative versions if they want to attach policies separate from the role definition.

Reply:
The warning messages were taken verbatim from the `iam_role` documentation page. https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/iam_role

If you want `exclusive` behaviour, you have to use _both_ `aws_iam_role_policy_attachments_exclusive` _and_ `aws_iam_role_policy_attachment` resources.

Reply:
Oh, in that case quoting the provider docs is fine